sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample15_MockingClientTypes.md

```diff
@@ -0,0 +1,1392 @@
+
+# Mocking Client Types
+
+Service Bus is built to support unit testing with mocks, as described in the [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking). This is an important feature of the library that allows developers to write tests that are completely focused on their own application logic, though they depend on Service Bus types.
+
+The following examples focus on scenarios likely to occur in applications, and demonstrate how to mock the Service Bus types typically used in each scenario. The code snippets utilize the mock object framework, Moq, in order to provide practical examples. However, many mocking frameworks exist and can be used with the same approach in mind.
+
+## Table of contents
+
+- **Sending and receiving messages**
+  - [Sending a message to a queue](#sending-messages-to-a-queue)
+  - [Receiving messages from a queue](#receiving-messages-from-a-queue)
+  - [Sending a batch of messages](#sending-a-batch-of-messages)
+  - [Sending and receiving messages using sessions](#session-sending-and-receiving)
+  - [Peeking a message](#peeking-a-message)
+- **Message Settlement**
+  - [Completing a message](#completing-a-message)
+  - [Deferring a message and receiving deferred messages](#deferring-a-message-and-receiving-deferred-messages)
+  - [Abandoning a messages](#abandoning-a-message)
+- **Testing the processor**
+  - [Testing the `ServiceBusProcessor` handlers](#for-the-servicebusprocessor)
+  - [Testing the `ServiceBusSessionProcessor` handlers](#for-the-servicebussessionprocessor)
+  - [Simulating the `ServiceBusProcessor` running](#for-the-servicebusprocessor-1)
+  - [Simulating the `ServiceBusSessionProcessor` running](#for-the-servicebussessionprocessor-1)
+- **CRUD operations**
+  - [Sending and receiving messages using topics and subscriptions](#sending-and-receiving-messages-using-topics-and-subscriptions)
```

Comment:
This seems weird to me.   I'd expect the CRUD section to be just using the admin client to manage the underlying resources.   Since we have sending/receiving in it's own section, would it make sense to break this into three snippets to follow the queue patterns?

- Creating topics and subscriptions. (CRUD)
- Sending to topics (Sending and receiving) 
- Receiving from a subscription (Sending and receiving)